typescript/sdk/src/deploy/core/HyperlaneCoreDeployer.ts

```diff
@@ -42,21 +41,56 @@ export class HyperlaneCoreDeployer<
     this.startingBlockNumbers = objMap(configMap, () => undefined);
   }
 
-  deployInterchainGasPaymaster<LocalChain extends Chain>(
+  async deployInterchainGasPaymaster<LocalChain extends Chain>(
     chain: LocalChain,
     proxyAdmin: ProxyAdmin,
     deployOpts?: DeployOptions,
-  ): Promise<
-    ProxiedContract<InterchainGasPaymaster, TransparentProxyAddresses>
-  > {
-    return this.deployProxiedContract(
+  ): Promise<ConnectionClientContracts> {
+    const owner = this.configMap[chain].owner;
+    const interchainGasPaymaster = await this.deployProxiedContract(
       chain,
       'interchainGasPaymaster',
       [],
       proxyAdmin,
-      [],
+      [owner],
+      deployOpts,
+    );
+    const interchainGasOverhead = await this.deployProxiedContract(
+      chain,
+      'interchainGasOverhead',
```

Comment:
Didn't @tkporter want to have the overhead contract to be at the existing IGP address?

Reply:
Yeah, imo this would be the smoothest transition for users

Reply:
I also wonder if we should rename the deployed overhead contract to be something like "multisigIsmIgp" and then to have the base / inner IGP be called something like "baseIgp". So that it's very clear what the use of each IGP is